### ISSUE TITLE
My application keeps getting flagged by McAfee after using botan

### ISSUE THREAD

**Issue by doctor-yes** [randombit/botan#1614]
My application keeps getting flagged by McAfee after using botan. Pretty sure it is related to botan as if I remove botan from the build, everything is fine. What could be the issue and how can I fix it? Thanks.

**Comment by randombit** [comment]
Sorry, no idea what the problem could be here. Does the error give some kind of malware identifier or other reasoning? One thing to try would be using a minimized build, on the assumption McAfee is triggering on some code construct or literal constant that it heuristically decides is malware-ish, if so perhaps restricting to just what you need the problematic code/constants/whatever will be skipped.

**Comment by randombit** [comment]
You could also try submitting the false positive binary as described here https://service.mcafee.com/webcenter/portal/cp/home/articleview?locale=en-US&articleId=TS102053

**Comment by doctor-yes** [comment]
Not too much information from McAfee, it just said "Malware Detected" and "because its reputation (Known Malicious) is below the configured Clean threshold." It also said "Threat name | Real Protect-LS!fc9a505015eb" if that means anything.

Regarding a minimized build, I used "--minimized-build" already when running the configure.py, is that what you mean by minimized build? Or I have to strip out things that I did not really call (any good way to do this automatically)?  

I'm working with McAfee, but they are unbelievably slow. I will add a note if they identify something so it can be looked into.

**Comment by randombit** [comment]
> Regarding a minimized build, I used "--minimized-build" already when running the configure.py, is that what you mean by minimized build?

Yes that was what I meant, the hope was McAfee dislikes something in botan that is unrelated to your application, and minimized build would just cause that code to be skipped entirely. Hm. Can you post your full `configure.py` line?

> I will add a note if they identify something so it can be looked into.

Please do. Obviously this is a very unfortunate situation. But lacking more information (like, what McAfee thinks is going on) I'm not sure what we can do from our side.

**Comment by doctor-yes** [comment]
python ./configure.py --cc=gcc --cpu=x86 --minimized-build --single-amalgamation-file --amalgamation --enable-modules=rsa,auto_rng,sha1,emsa1,hmac,win32_stats 

Are you aware of any other users who may run into a similar issue?

**Comment by randombit** [comment]
> Are you aware of any other users who may run into a similar issue?

No, this is the first time I have heard of this being a problem for anyone.

You might try removing `win32_stats` and adding `system_rng`. The win32 stats module does things that might seem malware-ish, such as iterating over the processes in the system. Sorry I would have suggested this earlier but when pondering what operations we perform that might trigger a malware warning I completely forgot about this code.

**Comment by doctor-yes** [comment]
That is it. After changing win32_stats to system_rng, McAfee is ok with it now. :)
